eng/references.targets

```diff
@@ -37,15 +37,13 @@
 
   <!-- Make shared framework assemblies not app-local (non private). -->
   <Target Name="UpdateProjectReferencesWithPrivateAttribute"
-          AfterTargets="AssignProjectConfiguration"
-          BeforeTargets="PrepareProjectReferences"
           Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp' and
-                     ('$(IsTestProject)' == 'true' or '$(IsTestSupportProject)' == 'true') and
-                     '@(ProjectReferenceWithConfiguration)' != ''">
+                     ('$(IsTestProject)' == 'true' or '$(IsTestSupportProject)' == 'true' or '$(IsSourceProject)' == 'true') and
+                     '@(ProjectReference)' != ''">
     <ItemGroup>
-      <ProjectReferenceWithConfiguration PrivateAssets="all"
-                                         Private="false"
-                                         Condition="$(NetCoreAppLibrary.Contains('%(Filename);'))" />
+      <ProjectReference PrivateAssets="all"
+                        Private="false"
+                        Condition="$(NetCoreAppLibrary.Contains('%(Filename);')) and '%(ProjectReference.DonotUseInboxVersion)' != 'true'" />
```

Comment:
Small nit: `DonotUseInboxVersion` might be easier to read if we name it `DontUseInboxVersion`